src/jvmMain/kotlin/mu/internal/KLoggerNameResolver.kt

```diff
@@ -33,18 +33,12 @@ internal object KLoggerNameResolver {
      * unwrap companion class to enclosing class given a Java Class
      */
     private inline fun <T : Any> unwrapCompanionClass(clazz: Class<T>): Class<*> {
-        if (clazz.enclosingClass != null) {
-            try {
-                val field = clazz.enclosingClass.getDeclaredField(clazz.simpleName)
-                if (Modifier.isStatic(field.modifiers) && field.type == clazz) {
-                    // && field.get(null) === obj
-                    // the above might be safer but problematic with initialization order
-                    return clazz.enclosingClass
-                }
-            } catch (e: Exception) {
-                //ok, it is not a companion object
-            }
-        }
-        return clazz
+        return clazz.enclosingClass?.let { enclosingClass ->
```

Comment:
Searching through `declaredFields` instead of using `getDeclaredField()` has the advantage of not throwing `NoSuchFieldException` if the declared field cannot be found. If there's a Kotlin-implemented version of this functionality that doesn't use reflection and handles the "field not found" case more gracefully than the Java-implemented version, I'd be glad to use that, but in general, it's better to avoid using exceptions as control flow when possible.